packages/mongo/collection/collection_extensions.js

```diff
@@ -0,0 +1,164 @@
+  },
+  
+  /**
+   * Add a prototype method to all collection instances
+   * Method is bound to the collection instance
+   */
+  addPrototypeMethod(name, method) {
+    if (typeof name !== 'string' || !name) {
+      throw new Error('Prototype method name must be a non-empty string');
+    }
+    if (typeof method !== 'function') {
+      throw new Error('Prototype method must be a function');
+    }
+    
+    // Check for reserved names (reserved for future hook system)
+    const reservedNames = ['before', 'after', 'direct', 'hookOptions', '_hooks'];
+    if (reservedNames.includes(name)) {
+      throw new Error(`Method name '${name}' is reserved for future hook system`);
+    }
+    
+    this._prototypeMethods.set(name, method);
+  },
+
+  /**
+   * Backwards compatibility alias for lai:collection-extensions
+   * @deprecated Use addPrototypeMethod instead
+   */
+  addPrototype(name, method) {
+    return this.addPrototypeMethod(name, method);
+  },
+  
+  /**
+   * Add a static method to the Mongo.Collection constructor
+   */
+  addStaticMethod(name, method) {
+    if (typeof name !== 'string' || !name) {
+      throw new Error('Static method name must be a non-empty string');
+    }
+    if (typeof method !== 'function') {
+      throw new Error('Static method must be a function');
+    }
+    
+    this._staticMethods.set(name, method);
+  },
+  
+  /**
+   * Remove an extension (useful for testing)
+   */
+  removeExtension(extension) {
+    const index = this._extensions.indexOf(extension);
+    if (index > -1) {
+      this._extensions.splice(index, 1);
+    }
+  },
+  
+  /**
+   * Remove a prototype method
+   */
+  removePrototypeMethod(name) {
+    this._prototypeMethods.delete(name);
+  },
+
+  /**
+   * Backwards compatibility alias for lai:collection-extensions
+   * @deprecated Use removePrototypeMethod instead
+   */
+  removePrototype(name) {
+    return this.removePrototypeMethod(name);
+  },
+  
+  /**
+   * Remove a static method
+   */
+  removeStaticMethod(name) {
+    this._staticMethods.delete(name);
+  },
+  
+  /**
+   * Clear all extensions (useful for testing)
+   */
+  clearExtensions() {
+    this._extensions.length = 0;
+    this._prototypeMethods.clear();
+    this._staticMethods.clear();
+  },
+  
+  /**
+   * Get all registered extensions (useful for debugging)
+   */
+  getExtensions() {
+    return [...this._extensions];
+  },
+  
+  /**
+   * Get all registered prototype methods (useful for debugging)
+   */
+  getPrototypeMethods() {
+    return new Map(this._prototypeMethods);
+  },
+  
+  /**
+   * Get all registered static methods (useful for debugging)
+   */
+  getStaticMethods() {
+    return new Map(this._staticMethods);
+  },
+  
+
+  
+  /**
+   * Apply all extensions to a collection instance
+   * Called during collection construction
+   */
+  _applyExtensions(instance, name, options) {
+    // Apply constructor extensions
+    this._extensions.forEach(extension => {
```

Comment:
I would prefer to see here (and other places) `for of` loopas it is more performant than `forEach`.